File: CityBlocksFlyover/cbf.rb
# city blocks flyover soln in ruby for codeeval by steven a dunn

File.open(ARGV[0], 'r') do |file|
    while line = file.gets
        x, y = line.split

        x = x[1..-2].split(",").map{|a| a.to_i}
        xs = x.zip(x[1..-1])[0..-2]

        y = y[1..-2].split(",").map{|a| a.to_i}
        ys = y.zip(y[1..-1])[0..-2]

        m = y[-1]/x[-1].to_f

        total = 0
        xs.each do |x|
            ys.each do |y|
                if ( !(m * x[0] >= y[1] || m * x[1] <= y[0]) )
                    total += 1
                end
            end
        end
        puts total
    end
end

File: CityBlocksFlyover/py3/cbf.py
# city blocks flyover soln in py3 for code eval by steven a dunn

import sys

f = open(sys.argv[1], 'r')
for line in f:
    x, y = line.rstrip().split(' ')

    x = list(map(int, x[1:-1].split(",")))
    y = list(map(int, y[1:-1].split(",")))

    xs = list(zip(x, x[1:]))
    ys = list(zip(y, y[1:]))

    m = y[-1]/float(x[-1])

    total = 0

    for i in xs:
        for j in ys:
            if not (m * i[0] >= j[1] or m * i[1] <= j[0]):
                total += 1
    print (total)
f.close()